handler/schedule.go

```diff
@@ -206,6 +233,80 @@ func (s *ScheduleHandler) getRandomSchedule(jobType k8upv1alpha1.JobType, origin
 }
 
 func (s *ScheduleHandler) setEffectiveSchedule(jobType k8upv1alpha1.JobType, schedule k8upv1alpha1.ScheduleDefinition) {
-	s.schedule.Status.EffectiveSchedules[jobType] = schedule
+	if s.effectiveSchedule == nil {
+		s.createNewEffectiveScheduleObj()
+	}
+	schedules := s.effectiveSchedule.Spec.EffectiveSchedules
+	schedules = append(schedules, k8upv1alpha1.JobRef{
+		Name:              s.schedule.Name,
+		Namespace:         s.schedule.Namespace,
+		JobType:           jobType,
+		GeneratedSchedule: schedule,
+	})
+	s.effectiveSchedule.Spec.EffectiveSchedules = schedules
+	s.requireStatusUpdate = true
+}
+
+func (s *ScheduleHandler) createNewEffectiveScheduleObj() {
+	newSchedule := &k8upv1alpha1.EffectiveSchedule{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cfg.Config.OperatorNamespace,
+			Name:      rand.String(32),
```

Comment:
For debugging purposes, I suggest to find a more meaningful naming convention than just 32 random characters, e.g. `random-weekly-$(RAND_5)` (i.e. the original schedule plus a short random suffix).

Reply:
I thought about it too. but it's not feasible:

1. If the only/last schedule changes from `@weekly-random` to `@daily-random` then it's even more confusing. We would have to delete the old, and then create a new object. Sounds like overengineering just for naming the users shouldn't care about much?
2. The `Schedule` can specify a `check` schedule and `backup` schedule with different `@...-random` schedules, but both will make it into the same `EffectiveSchedule` currently. So which one to pick? 

Thinking about the 2nd point. There might be a flaw in the current code if both job types end up in the same effective schedule object. especially later for deduplication maybe it's better to have all backup schedules in one, all Checks in another... I need to look into it.

Reply:
After a 1-on-1 discussion we concluded that there should be an effective Schedule resource for each Job type, not per schedule itself. Otherwise this would be harder to deduplicate for different job types.